src/NServiceBus.AcceptanceTests/Outbox/When_headers_contain_special_characters.cs

```diff
@@ -0,0 +1,112 @@
+﻿namespace NServiceBus.AcceptanceTests.Outbox
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using EndpointTemplates;
+    using NServiceBus.Pipeline;
+    using NUnit.Framework;
+
+    public class When_headers_contain_special_characters : NServiceBusAcceptanceTest
+    {
+        static Guid messageId = Guid.NewGuid();
+        static bool SavedOutBoxRecord = false;
+        static Dictionary<string, string> sentHeaders = new Dictionary<string, string>
+        {
+            { "a-B1", "a-B" },
+            { "a-B2", "a-ɤϡ֎ᾣ♥-b" },
+            { "a-ɤϡ֎ᾣ♥-B3", "a-B" },
+            { "a-B4", "a-\U0001F60D-b" },
+            { "a-\U0001F605-B5", "a-B" },
+            { "a-B6", "a-😍-b" },
+            { "a-😅-B7", "a-B" },
+            {"a-b8", "奥曼克"},
+            {"a-B9", "٩(-̮̮̃-̃)۶ ٩(●̮̮̃•̃)۶ ٩(͡๏̯͡๏)۶ ٩(-̮̮̃•̃)" },
+            {"a-b10", "தமிழ்" }
+        };
+
+        [Test]
+        public async Task Outbox_should_work()
+        {
+            Requires.OutboxPersistence();
+
+            var context =
+                await Scenario.Define<Context>()
+                    .WithEndpoint<OutboxEndpoint>(b => b.When(session => session.SendLocal(new PlaceOrder())))
+                    .Done(c => c.MessageReceived)
+                    .Run(TimeSpan.FromSeconds(20));
+
+            Assert.IsNotEmpty(context.UnicodeHeaders);
+            CollectionAssert.IsSubsetOf(sentHeaders, context.UnicodeHeaders);
+        }
+
+        class Context : ScenarioContext
+        {
+            public Dictionary<string, string> UnicodeHeaders { get; set; }
+            public bool MessageReceived { get; set; }
+        }
+
+        public class OutboxEndpoint : EndpointConfigurationBuilder
+        {
+            public OutboxEndpoint()
+            {
+                EndpointSetup<DefaultServer>(b =>
+                {
+                    b.EnableOutbox();
+                    b.Pipeline.Register("BlowUpBeforeDispatchBehavior", new BlowUpBeforeDispatchBehavior(), "Force reading the message from Outbox storage.");
+                    b.Recoverability().Immediate(a => a.NumberOfRetries(1));
+                });
+            }
+            class BlowUpBeforeDispatchBehavior : IBehavior<IBatchDispatchContext, IBatchDispatchContext>
+            {
+                public async Task Invoke(IBatchDispatchContext context, Func<IBatchDispatchContext, Task> next)
+                {
+                    if (!SavedOutBoxRecord)
```

Comment:
I have doubts if that's really accessible from here. I though I was just too tired on Friday, but still I don't see how that can be accessed. Looking at usages, I don't see any example of it being used in the context of tests and when testing with Nhibernate it can't be found. So I'd rather stick to the flag.

<img width="645" alt="usages" src="https://cloud.githubusercontent.com/assets/7335079/26295846/525bcb6a-3ecc-11e7-843c-73a8137ef7e6.png">